cas_implementations/src/cas/pickle.rs

```diff
@@ -0,0 +1,102 @@
+use holochain_core_types::{
+    cas::{
+        content::{Address, AddressableContent, Content},
+        storage::ContentAddressableStorage,
+    },
+    error::HolochainError,
+};
+use pickledb::{PickleDb, PickleDbDumpPolicy, SerializationMethod};
+use std::{
+    fmt::{Debug, Error, Formatter},
+    path::Path,
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+use uuid::Uuid;
+
+const PERSISTENCE_INTERVAL: Duration = Duration::from_millis(5000);
+
+#[derive(Clone)]
+pub struct PickleStorage {
+    id: Uuid,
+    db: Arc<Mutex<PickleDb>>,
+}
+
+impl Debug for PickleStorage {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
+        f.debug_struct("PickleStorage")
+            .field("id", &self.id)
+            .finish()
+    }
+}
+
+impl PickleStorage {
+    pub fn new<P: AsRef<Path>>(db_path: P) -> PickleStorage {
+        PickleStorage {
+            id: Uuid::new_v4(),
+            db: Arc::new(Mutex::new(PickleDb::new(
```

Comment:
Why did we choose mutex over Rwlock?

Reply:
No reason

Reply:
Nice was just checking. I think we should use Rwlock just so that it allows multiple people to read vs locking on the read. What do you think?